src/Umbraco.Infrastructure/Persistence/Repositories/Implement/ContentRepositoryBase.cs

```diff
@@ -1106,14 +1110,16 @@ protected void PersistRelations(TEntity entity)
                     .WhereIn<NodeDto>(x => x.UniqueId, guids);
             }).ToDictionary(x => x.UniqueId, x => x.NodeId);
 
-            // Lookup all relation type IDs
-            var relationTypeLookup = RelationTypeRepository.GetMany(Array.Empty<int>()).ToDictionary(x => x.Alias, x => x.Id);
-
             // Get all valid relations
-            var relations = new List<ReadOnlyRelation>(references.Count);
+            var relations = new List<(int ChildId, int RelationTypeId)>(references.Count);
             foreach (UmbracoEntityReference reference in references)
             {
-                if (!automaticRelationTypeAliases.Contains(reference.RelationTypeAlias))
+                if (string.IsNullOrEmpty(reference.RelationTypeAlias))
+                {
+                    // Reference does not specify a relation type alias, so skip adding a relation
+                    Logger.LogDebug("The reference to {Udi} does not specify a relation type alias, so it will not be saved as relation.", reference.Udi);
+                }
```

Comment:
We don't want to log an error/warning when a reference doesn't specify a relation type alias, which can happen when using the `UmbracoEntityReference(Udi udi)` overload with an UDI/entity type isn't a document or media...